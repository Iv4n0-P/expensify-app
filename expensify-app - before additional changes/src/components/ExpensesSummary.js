import React from 'react'
import selectExpensesTotal from '../selectors/expenses-total'
import SelectExpenses from '../selectors/expenses'
import { connect } from 'react-redux'
import numeral from 'numeral'
import { Link } from 'react-router-dom'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formatedExpensesTotal}</span></h1>
                <div className="page-header__actions">
                    <Link className="button-blue" to="/create">Add Expense</Link>
                </div>
             </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expensesToShow = SelectExpenses(state.expenses, state.filters)
    return {
        expensesTotal: selectExpensesTotal(expensesToShow),
        expenseCount: expensesToShow.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary)


