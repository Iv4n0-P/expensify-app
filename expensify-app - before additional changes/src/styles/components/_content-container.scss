.content-container {
    max-width: 80rem;
    margin: $l-size auto;
    padding: 0 $m-size;
}