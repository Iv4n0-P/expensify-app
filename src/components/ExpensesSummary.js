import React from 'react'
import selectExpensesTotal from '../selectors/expenses-total'
import SelectExpenses from '../selectors/expenses'
import { connect } from 'react-redux'
import numeral from 'numeral'
import { Link } from 'react-router-dom'

export const ExpensesSummary = ({ expenseCount, expensesTotal, hiddenExpenses, allExpenses }) => {
    const expenseWord = allExpenses === 1 ? 'expense' : 'expenses'
    const expenseWord2 = hiddenExpenses === 1 ? 'expense' : 'expenses'
    const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00')

    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> of {allExpenses} {expenseWord} totalling <span>{formatedExpensesTotal}</span></h1>
                <p>{hiddenExpenses !==0 && `${hiddenExpenses} hidden ${expenseWord2}`}</p>
                <div className="page-header__actions">
                    <Link className="button-blue" to="/create">Add Expense</Link>
                </div>
             </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const expensesToShow = SelectExpenses(state.expenses, state.filters)
    return {
        expensesTotal: selectExpensesTotal(expensesToShow),
        expenseCount: expensesToShow.length,
        allExpenses: state.expenses.length,
        hiddenExpenses: state.expenses.length - expensesToShow.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary)


