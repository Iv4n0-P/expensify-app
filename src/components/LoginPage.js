import React from 'react'
import { connect } from 'react-redux'
import { startLogin, startLoginFacebook, startLoginTwitter, startLoginGithub } from '../actions/auth'

export const LoginPage = ({ startLogin, startLoginFacebook, startLoginTwitter, startLoginGithub }) => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control</p>
            <button id='startLogin' className="button-blue button-google" onClick={startLogin}>Login with Google</button>
            <button id='startLoginFacebook' className="button-blue button-facebook" onClick={startLoginFacebook}>Login with Facebook</button>
            <button id='startLoginTwitter' className="button-blue button-twitter" onClick={startLoginTwitter}>Login with Twitter</button>
            <button id='startLoginGithub' className="button-blue button-github" onClick={startLoginGithub}>Login with Github</button>
        </div>
    </div>
)

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin()),
    startLoginFacebook: () => dispatch(startLoginFacebook()),
    startLoginTwitter: () => dispatch(startLoginTwitter()),
    startLoginGithub: () => dispatch(startLoginGithub())
})

export default connect(undefined, mapDispatchToProps)(LoginPage)