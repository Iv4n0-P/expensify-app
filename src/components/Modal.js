import React from 'react'
import Modal from 'react-modal'

export const OptionModal = (props) => (

  <Modal
    isOpen={props.showModal}
    contentLabel="Confirm deletion"
    ariaHideApp={false}
    className='modal'
  >
    <div>
      <p>Do you want to delete this expense?</p>
      <button className="button-blue button-github" onClick={props.onRemove}>Remove Expense</button>
      <button className="button-blue button-github" onClick={props.handleCloseModal}>Close</button>
    </div>
  </Modal>
)