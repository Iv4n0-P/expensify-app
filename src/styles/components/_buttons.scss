.button-blue {
    padding: $s-size $xl-size;
    background: $blue;
    color: white;
    display: inline-block;
    border-width: 0px;
    //ili border: none;
    font-size: $font-size-large; //čisto font u varijablu se može isto
    text-decoration: none;
    line-height: 1;
}

.button-blue--nobg {
    background: none;
    border: none;
    color: #f7f7f7;
}

.button-blue--secondary {
    background: #888888;
}