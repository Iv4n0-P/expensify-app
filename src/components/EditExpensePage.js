import React from 'react'
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'
import { OptionModal } from './Modal'

export class EditExpensePage extends React.Component {

  state = {
    selectedOption: undefined
  }

  handleOpenModal = () => {
    this.setState(() => ({ selectedOption: true }));
  }

  handleCloseModal = () => {
    this.setState(() => ({ selectedOption: undefined }));
  }

  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  };

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Edit Expense</h1>
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            expense={this.props.expense}
            onSubmit={this.onSubmit}
          />
          <button className="button-blue button-blue--secondary" onClick={this.handleOpenModal}>Remove Expense</button>

          <OptionModal
          showModal={this.state.selectedOption}
          onRemove={this.onRemove}
          handleCloseModal={this.handleCloseModal}
          />
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)




