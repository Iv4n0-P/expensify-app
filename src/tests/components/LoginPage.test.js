import { LoginPage } from '../../components/LoginPage'
import React from 'react'
import {shallow} from 'enzyme'

test ('should render LoginPage correctly', () => {
    const wrapper = shallow(<LoginPage />)
    expect(wrapper).toMatchSnapshot()
 })

test('should call startLogin on button click', () => {
    const startLogin = jest.fn()
    const wrapper = shallow(<LoginPage startLogin={startLogin} />)
    wrapper.find('#startLogin').simulate('click')
    expect(startLogin).toHaveBeenCalled()
})

test('should call startLoginFacebook on button click', () => {
    const startLoginFacebook = jest.fn()
    const wrapper = shallow(<LoginPage startLoginFacebook={startLoginFacebook} />)
    wrapper.find('#startLoginFacebook').simulate('click')
    expect(startLoginFacebook).toHaveBeenCalled()
})

test('should call startLoginTwitter on button click', () => {
    const startLoginTwitter = jest.fn()
    const wrapper = shallow(<LoginPage startLoginTwitter={startLoginTwitter} />)
    wrapper.find('#startLoginTwitter').simulate('click')
    expect(startLoginTwitter).toHaveBeenCalled()
})

test('should call startLoginGithub on button click', () => {
    const startLoginGithub = jest.fn()
    const wrapper = shallow(<LoginPage startLoginGithub={startLoginGithub} />)
    wrapper.find('#startLoginGithub').simulate('click')
    expect(startLoginGithub).toHaveBeenCalled()
})