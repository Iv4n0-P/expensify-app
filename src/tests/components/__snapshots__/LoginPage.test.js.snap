// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LoginPage correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className="box-layout__title"
    >
      Expensify
    </h1>
    <p>
      It's time to get your expenses under control
    </p>
    <button
      className="button-blue button-google"
      id="startLogin"
    >
      Login with Google
    </button>
    <button
      className="button-blue button-facebook"
      id="startLoginFacebook"
    >
      Login with Facebook
    </button>
    <button
      className="button-blue button-twitter"
      id="startLoginTwitter"
    >
      Login with Twitter
    </button>
    <button
      className="button-blue button-github"
      id="startLoginGithub"
    >
      Login with Github
    </button>
  </div>
</div>
`;
